stl/inc/span

```diff
@@ -0,0 +1,665 @@
+        return _Right.end();
+    }
+#endif // __cpp_lib_concepts
+
+    _NODISCARD constexpr const_iterator cbegin() const noexcept {
+        return begin();
+    }
+
+    _NODISCARD constexpr const_iterator cend() const noexcept {
+        return end();
+    }
+
+    _NODISCARD constexpr reverse_iterator rbegin() noexcept {
+        return reverse_iterator{end()};
+    }
+
+    _NODISCARD constexpr reverse_iterator rend() noexcept {
+        return reverse_iterator{begin()};
+    }
+
+    _NODISCARD constexpr const_reverse_iterator rbegin() const noexcept {
+        return const_reverse_iterator{end()};
+    }
+
+    _NODISCARD constexpr const_reverse_iterator rend() const noexcept {
+        return const_reverse_iterator{begin()};
+    }
+
+    _NODISCARD constexpr const_reverse_iterator crbegin() const noexcept {
+        return rbegin();
+    }
+
+    _NODISCARD constexpr const_reverse_iterator crend() const noexcept {
+        return rend();
+    }
+
+    _NODISCARD constexpr pointer _Unchecked_begin() noexcept {
+        return _Mydata;
+    }
+
+    _NODISCARD constexpr const_pointer _Unchecked_begin() const noexcept {
+        return _Mydata;
+    }
+
+    _NODISCARD constexpr pointer _Unchecked_end() noexcept {
+        return _Mydata + size();
+    }
+
+    _NODISCARD constexpr const_pointer _Unchecked_end() const noexcept {
+        return _Mydata + size();
+    }
+
+private:
+    pointer _Mydata{nullptr};
+};
+
+// [span.tuple] Tuple interface
+template <class _Ty, size_t _Extent>
+struct tuple_size<span<_Ty, _Extent>> : integral_constant<size_t, _Extent> {
+    static_assert(dynamic_extent != _Extent, "std::span<T, dynamic_extent> is not tuple-like. ");
+};
+
+template <size_t _Index, class _Ty, size_t _Extent>
+struct tuple_element<_Index, span<_Ty, _Extent>> {
+    static_assert(dynamic_extent != _Extent, "std::span<T, dynamic_extent> is not tuple-like. ");
+    static_assert(_Index < _Extent, "Index out of bounds for a std::span of this extent. ");
+    using type = _Ty;
+};
+
+template <size_t _Index, class _Ty, size_t _Extent>
+_NODISCARD constexpr _Ty& get(span<_Ty, _Extent> _Span) noexcept {
+    static_assert(dynamic_extent != _Extent, "std::get<> not supported for std::span<T, dynamic_extent>");
+    static_assert(_Index < _Extent, "Index out of bounds for a std::span of this extent. ");
+    return _Span[_Index];
+}
+
+#if _HAS_STD_BYTE
+// [span.objectrep] Views of object representation
+// FUNCTION TEMPLATE as_bytes
+template <class _Ty, size_t _Extent, enable_if_t<_Extent <= dynamic_extent / sizeof(_Ty), int> = 0>
+_NODISCARD auto as_bytes(span<_Ty, _Extent> _Sp) noexcept -> span<const byte, sizeof(_Ty) * _Extent> {
+    return {reinterpret_cast<const byte*>(_Sp.data()), _Sp.size_bytes()};
+}
+
+template <class _Ty>
+_NODISCARD auto as_bytes(span<_Ty, dynamic_extent> _Sp) noexcept -> span<const byte, dynamic_extent> {
+    return {reinterpret_cast<const byte*>(_Sp.data()), _Sp.size_bytes()};
+}

```

Comment:
This is depicted as a single overload in the Standard. I see that you're guarding against compile-time integer overflow, which is not depicted in the Standard.

I believe that this should be a single overload, following the Standard's "equivalent to" implementation. If you believe that detecting compile-time integer overflow is important, it seems like that should be a `static_assert` instead, and there should be an LWG issue filed for this.

In general, if you feel you have to deviate significantly from what is depicted in the Standard, this should be clearly called out in your PR notes.

Reply:
Yeah, I moved the check into `size_bytes`